### PR TITLE
Adding collision metrics for the closed-loop evaluator

### PR DESCRIPTION
**Pull request by perone**
This PR will add the collision metrics, the tests also depend on the custom mocked data.
I'm using here the `simulation_output.recorded_agents` but I'm not sure about what should come in this array, so I'll submit this PR and then when we have data coming in I can change and fix it.

### CODE REVIEW

File: l5kit/l5kit/cle/metrics.py [hunk 1]
```diff
@@ -23,6 +25,84 @@ def compute(self, simulation_output: SimulationOutput) -> torch.Tensor:
...
+        :param observed_frame: the ground-truth frame
+        :param simulated_frame_ego_state: ego state from the simulated frame,
+                                          this is a 1D array with the frame
+                                          ego state
+        :returns: metric result for the frame, where 1 means a collision,
+                  and 0 otherwise.
+        """
+        simulated_centroid = simulated_frame_ego_state[:TrajectoryStateIndices.Y + 1].cpu().numpy()
+        simulated_angle = simulated_frame_ego_state[TrajectoryStateIndices.THETA].cpu().numpy()
+        simulated_extent = np.r_[EGO_EXTENT_LENGTH, EGO_EXTENT_WIDTH]
```
Comment (lucabergamini) [inline l5kit/l5kit/cle/metrics.py:50]: this is the same as `np.asarray([EGO_EXTENT_LENGTH, EGO_EXTENT_WIDTH])` right?

File: l5kit/l5kit/cle/metrics.py [hunk 2]
```diff
@@ -23,6 +25,84 @@ def compute(self, simulation_output: SimulationOutput) -> torch.Tensor:
...
+        simulated_scene_ego_state = simulation_output.simulated_ego_states
+        simulated_agents = simulation_output.simulated_agents
+
+        if len(simulated_agents) < len(simulated_scene_ego_state):
+            raise ValueError("More simulated timesteps than observed.")
+
+        num_frames = simulated_scene_ego_state.size(0)
+        metric_results = torch.zeros(num_frames, device=simulated_scene_ego_state.device)
+        for frame_idx in range(num_frames):
+            recorded_agent_frame = simulated_agents[frame_idx]
```
Comment (lucabergamini) [inline l5kit/l5kit/cle/metrics.py:75]: lol naming here could be improved; unless this is here to test if I'm paying attention while reviewing xD

**Review by lucabergamini — COMMENTED** [review]
1 minor

File: l5kit/l5kit/cle/metrics.py (hunk 1 above)
Comment (perone) [reply l5kit/l5kit/cle/metrics.py:50]: Yes, it is just a short hand to look closer to how lists are built in Python.

File: l5kit/l5kit/cle/metrics.py (hunk 2 above)
Comment (perone) [reply l5kit/l5kit/cle/metrics.py:75]: haha will fix it

**Review by lucabergamini — APPROVED** [review]

**Pull request merged**